src/abacus-ts/calculators/orders.ts

```diff
@@ -0,0 +1,196 @@
+    case OrderStatus.Canceling:
+      return OrderStatus.Open;
+    case OrderStatus.Canceled:
+    case OrderStatus.PartiallyCanceled:
+      return OrderStatus.Canceled;
+    case OrderStatus.Filled:
+      return OrderStatus.Filled;
+    default:
+      assertNever(status);
+      // should never happen since we made OrderStatus manually
+      return OrderStatus.Open;
+  }
+}
+
+function maybeUpdateOrderIfExpired(
+  order: SubaccountOrder,
+  height: HeightResponse
+): SubaccountOrder {
+  if (order.status == null) {
+    return order;
+  }
+  // todo: why not handle Open?
+  if (
+    ![OrderStatus.Pending, OrderStatus.Canceling, OrderStatus.PartiallyFilled].includes(
+      order.status
+    )
+  ) {
+    return order;
+  }
+
+  // Check if order has expired based on goodTilBlock
+  if (order.goodTilBlock && order.goodTilBlock !== 0 && height.height >= order.goodTilBlock) {
+    let status = OrderStatus.Canceled;
+
+    // Check for partial fills
+    if (order.totalFilled != null && order.totalFilled.gt(0)) {
+      const remainingSize = order.size.minus(order.totalFilled);
+      if (order.totalFilled.gt(0) && remainingSize.gt(0)) {
+        status = OrderStatus.PartiallyCanceled;
+      }
+    }
+
+    return {
+      ...order,
+      status,
+      updatedAtMilliseconds: new Date(height.time).valueOf(),
+      updatedAtHeight: height.height,
+    };
+  }
+
+  return order;
+}
+
+function calculateBaseOrderStatus(order: IndexerCompositeOrderObject): OrderStatus | undefined {
+  const status = order.status;
+  if (status == null) {
+    return undefined;
+  }
+
+  if (status === IndexerBestEffortOpenedStatus.BESTEFFORTOPENED) {
+    return OrderStatus.Pending;
+  }
+
+  // Calculate filled amounts
+  const size = MustBigNumber(order.size);
+  const totalFilled = MustBigNumber(order.totalFilled);
+  const remainingSize = size.minus(totalFilled);
+  const hasPartialFill = totalFilled.gt(0) && remainingSize.gt(0);
+
+  // Handle partial fills
+  if (hasPartialFill) {
+    if (status === IndexerOrderStatus.OPEN) {
+      return OrderStatus.PartiallyFilled;
+    }
+    if (status === IndexerOrderStatus.CANCELED) {
+      return OrderStatus.PartiallyCanceled;
+    }
+  }
+
+  // Handle short-term order edge case
+  const isShortTermOrder = order.orderFlags === '0';
+  const isBestEffortCanceled = status === IndexerOrderStatus.BESTEFFORTCANCELED;
+  const isUserCanceled =
+    order.removalReason === 'USER_CANCELED' ||
+    order.removalReason === 'ORDER_REMOVAL_REASON_USER_CANCELED';
+
+  if (isShortTermOrder && isBestEffortCanceled && !isUserCanceled) {
+    return OrderStatus.Pending;
+  }
+
+  // Direct mapping for remaining cases
+  switch (status) {
+    case IndexerOrderStatus.OPEN:
+      return OrderStatus.Open;
+    case IndexerOrderStatus.FILLED:
+      return OrderStatus.Filled;
+    case IndexerOrderStatus.CANCELED:
+      return OrderStatus.Canceled;
+    case IndexerOrderStatus.BESTEFFORTCANCELED:
```

Comment:
omg why is this enum not snake cased hahaha

Reply:
is this an auto-gen?

Reply:
yeah it's generated :/